packages/vector_graphics/lib/src/listener.dart

```diff
@@ -95,19 +117,26 @@ class FlutterVectorGraphicsListener extends VectorGraphicsCodecListener {
   final TextDirection? _textDirection;
 
   final ui.PictureRecorder _recorder;
-  final ui.Canvas _canvas;
+  ui.Canvas _canvas;
 
   final List<ui.Paint> _paints = <ui.Paint>[];
   final List<ui.Path> _paths = <ui.Path>[];
   final List<ui.Shader> _shaders = <ui.Shader>[];
   final List<_TextConfig> _textConfig = <_TextConfig>[];
   final List<Future<ui.Image>> _pendingImages = <Future<ui.Image>>[];
   final Map<int, ui.Image> _images = <int, ui.Image>{};
-
+  final List<Future<ui.ImageShader>> _pendingPatterns =
+      <Future<ui.ImageShader>>[];
+  final Map<int, ui.ImageShader> _patterns = <int, ui.ImageShader>{};
+  final Map<int, ui.Canvas> _patternCanvases = <int, ui.Canvas>{};
+  final Map<int, ui.PictureRecorder> _patternRecorders =
+      <int, ui.PictureRecorder>{};
```

Comment:
It might simplify this code a bit if instead of having N maps per property you want to track, you instead added one map that contains an object that had all of these properties:

Something like this:

```dart
class PatternState {
  ui.Canvas? canvas;
  ui.ImageShader? shader;
  ui.PictureRecorder? recorder
}
```

Reply:
done.